samples/bluetooth/hci_pwr_ctrl/src/main.c

```diff
@@ -0,0 +1,318 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2019 Andrei Stoica
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <sys/printk.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/hci_vs.h>
+
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/services/hrs.h>
+
+static struct bt_conn *default_conn;
+static u16_t default_conn_handle;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID16_ALL, 0x0d, 0x18),
+};
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+#define DEVICE_BEACON_TXPOWER_NUM  8
+
+static struct k_thread pwr_thread_data;
+static K_THREAD_STACK_DEFINE(pwr_thread_stack, 320);
+
+static const s8_t txp[DEVICE_BEACON_TXPOWER_NUM] = {4, 0, -3, -8,
+						    -15, -18, -23, -30};
+static const struct bt_le_adv_param *param = BT_LE_ADV_PARAM
+	(BT_LE_ADV_OPT_CONNECTABLE | BT_LE_ADV_OPT_USE_NAME,
+	0x0020, 0x0020);
+
+static void read_conn_rssi(u16_t handle, s8_t *rssi)
+{
+	struct bt_hci_cp_read_rssi *cp;
+	struct bt_hci_rp_read_rssi *rp;
+	struct net_buf *buf, *rsp;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_READ_RSSI, sizeof(*cp));
+	if (!buf) {
+		printk("Unable to allocate command buffer\n");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_RSSI, buf, &rsp);
+	if (err) {
+		printk("RSSI HCI err: %d\n", err);
+	}
+
+	rp = (void *)rsp->data;
+	if (rp->status) {
+		printk("HCI cmd err: %d\n", rp->status);
+	} else {
+		*rssi = rp->rssi;
+	}
+
+	net_buf_unref(rsp);
+}
+
+
+static void set_tx_power(u8_t handle_type, u16_t handle, s8_t tx_pwr_lvl)
+{
+	struct bt_hci_cp_vs_write_tx_power_level *cp;
+	struct bt_hci_rp_vs_write_tx_power_level *rp;
+	struct net_buf *buf, *rsp;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_VS_WRITE_TX_POWER_LEVEL,
+				sizeof(*cp));
+	if (!buf) {
+		printk("Unable to allocate command buffer\n");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+	cp->handle_type = handle_type;
+	cp->tx_power_level = tx_pwr_lvl;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_WRITE_TX_POWER_LEVEL,
+				   buf, &rsp);
+	if (err) {
+		printk("TXP SET HCI err: %d\n", err);
```

Comment:
```suggestion
		printk("TXP SET HCI err: %d\n", err);
		return;
```

Reply:
Fixed for all HCI, i.e. [setter](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/787709a21828771d56ba998996e9a82646846617#diff-b9bd630d9245e9d01d535cbaa393a233R106), [getter](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/787709a21828771d56ba998996e9a82646846617#diff-b9bd630d9245e9d01d535cbaa393a233R143) and [rssi](https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/787709a21828771d56ba998996e9a82646846617#diff-b9bd630d9245e9d01d535cbaa393a233R69), again forgot about the cmd_send buffer handling.